jep/0000/README.adoc

```diff
@@ -0,0 +1,341 @@
+
+===== Suspended plugins
+
+Some plugins were found which were suspended when Jenkins 2 was released.
+These will not be updated as part of this effort.
+
+* https://plugins.jenkins.io/svn-release-mgr suspended since https://issues.jenkins-ci.org/browse/INFRA-2487
+* https://github.com/jenkinsci/cpptest-plugin suspended since https://issues.jenkins-ci.org/browse/INFRA-2487
+* https://github.com/jenkinsci/tfs-plugin suspended since https://issues.jenkins-ci.org/browse/INFRA-2751
+* https://github.com/jenkinsci/cflint-plugin (link:https://github.com/jenkinsci/CFLint-plugin/pull/3[PR-3]) suspended since https://issues.jenkins-ci.org/browse/INFRA-2751
+* https://github.com/jenkinsci/script-scm-plugin SECURITY-461
+* https://github.com/jenkinsci/rtc-plugin (superseded by team-concert)
+
+
+===== Never released plugins
+
+Some plugins were found which were suspended when Jenkins 2 was released.
+These will not be updated as part of this effort.
+
+* https://github.com/jenkinsci/cocoemma-plugin
+* https://github.com/jenkinsci/jwsdp-sqe-plugin
+* https://github.com/jenkinsci/pucm-plugin
+* https://github.com/jenkinsci/purecm-plugin
+
+
+== Motivation
+
+The link:https://commons.apache.org/proper/commons-digester/[Apache Commons Digester] that was included as a dependency of Jenkins Core is old and poorly maintained.
+It contains security vulnerabilities and also is not actually used by any part of Jenkins Core itself.
+Jenkins Core exposes an API to plugin that mitigates some of these security issues, but it is not the job of Jenkins to do this.
+Instead, we should remove the Digester dependency from Jenkins Core and the plugins that actually use Digester should handle these security concerns themselves.
+Also, because it is a dependency of Core, some plugins have taken a dependency on Digester being provided by Jenkins rather than using the API provided by Jenkins.
+
+Plugins in either of these scenarios will need to be updated or they will break after this removal.
+
+== Reasoning
+
+=== Risks of complete removal
+
+We chose to remove Digester from Jenkins Core in a manner that causes plugins which depend on it to fail unless they are updated.
+We made this choice based on an analysis of all plugins which are currently distributed by the Jenkins Update Center or which are
+in a repository in the link:https://github.com/jenkinsci[`jenkinsci` org on GitHub].
+All but a few of the plugins effected by this change are some combination of unmaintained, rarely used, and no longer or never published.
+We submitted PRs to all effected plugins.
+
+
+=== Detached plugin to maintain compatibility
+
+The current plan causes plugins which depend on Jenkins to provide Digester to fail unless they are updated.
+This could be mitigated by moving this dependency to a detached plugin.
+We decided against creating a detached pluging because there were a small number effected plugins and only a few of them have significant install base.
```

Comment:
```suggestion
We decided against creating a detached pluging because there were a small number of affected plugins and only a few of them have significant install base.
```